recipes/cgns/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+required_conan_version = ">=1.29.1"
+
+
+class CgnsConan(ConanFile):
+    name = "cgns"
+    description = "Standard for data associated with the numerical solution " \
+                  "of fluid dynamics equations."
+    topics = ("conan", "cgns", "data", "cfd", "fluids")
+    homepage = "http://cgns.org/"
+    license = "Zlib"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_hdf5": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_hdf5": True
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_hdf5:
+            self.requires("hdf5/1.12.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("CGNS-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["CGNS_BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["CGNS_ENABLE_HDF5"] = self.options.with_hdf5
+        self._cmake.configure()
+
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("license.txt", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        for binary in os.listdir(os.path.join(self.package_folder, "bin")):
+            if not binary.endswith(".dll"):
+                os.remove(os.path.join(self.package_folder, "bin", binary))
+
+        os.remove(os.path.join(self.package_folder, "include", "cgnsBuild.defs"))
+
+    def package_info(self):
```

Comment:
`USE_DLL` (too generic unfortunately) should be defined if Windows and shared: https://github.com/CGNS/CGNS/blob/341b4dfe413eff2aa3ccf9f48cc23fdc0cc999e6/src/cgnslib.h#L60